tools/codegen/core/gen_settings_ids.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python2.7
```

Comment:
Do we have a test or build rule or something that makes sure that the generated files are not updated manually?